File: Patterns/Command Pattern/CSharp/Command/GarageDoor/GarageDoor.cs
using System;

namespace Command.GarageDoor
{
    public class GarageDoor
    {
        public int State { get; private set; }

        public const int ON = 1;
        public const int OFF = 0;

        public GarageDoor()
        {
            State = OFF;
        }

        public void Up()
        {
            State = ON;
            Console.WriteLine("차고 문이 올라갑니다.");
        }

        public void Down()
        {
            State = OFF;
            Console.WriteLine("차고 문이 내려갑니다.");
        }

        public void Stop() => Console.WriteLine("차고 문이 멈췄습니다.");

        public void LightOn() => Console.WriteLine("차고 조명이 켜졌습니다.");

        public void LightOff() => Console.WriteLine("차고 조명이 꺼졌습니다.");
    }
}

File: Patterns/Command Pattern/Cpp/Command/GarageDoor/GarageDoor.cpp
#pragma once

#include <iostream>
#include <string>

using namespace std;

class GarageDoor {
private:
    int state;

public:
    const int ON = 1;
    const int OFF = 0;

    GarageDoor() : state{OFF} {}

    void up() {
        state = ON;
        cout << "차고 문이 올라갑니다.\n"; 
    }

    void down() {
        state = OFF;
        cout << "차고의 문이 내려갑니다.\n"; 
    }

    void stop() { 
        cout << "차고의 문이 멈췄습니다.\n"; 
    }

    void lightOn() { 
        cout << "차고의 조명이 켜졌습니다.\n"; 
    }

    void lightOff() { 
        cout << "차고의 조명이 꺼졌습니다.\n"; 
    }

    int getState() { return state; }
};